modules/graphics/weston.nix

```diff
@@ -12,137 +12,63 @@ in {
+      environment.systemPackages = with pkgs; [
+        weston
+      ];
+
+      # Next 2 services/targets are taken from official weston documentation:
+      # https://wayland.pages.freedesktop.org/weston/toc/running-weston.html
+
+      # Weston socket
+      systemd.user.sockets."weston" = {
+        unitConfig = {
+          Description = "Weston, a Wayland compositor";
+          Documentation = "man:weston(1) man:weston.ini(5)";
+        };
+        socketConfig = {
+          ListenStream = "%t/wayland-0";
+        };
+        wantedBy = ["weston.service"];
       };
-      wantedBy = ["multi-user.target"];
-    };
 
-    # systemd service for seatd
-    systemd.services."seatd" = {
-      description = "Seat management daemon";
-      documentation = ["man:seatd(1)"];
-      serviceConfig = {
-        Type = "simple";
-        ExecStart = "${pkgs.seatd}/bin/seatd -g video";
-        Restart = "always";
-        RestartSec = "1";
+      # Weston service
+      systemd.user.services."weston" = {
+        enable = true;
+        description = "Weston, a Wayland compositor, as a user service TEST";
+        documentation = ["man:weston(1) man:weston.ini(5)" "https://wayland.freedesktop.org/"];
+        requires = ["weston.socket"];
+        after = ["weston.socket" "ghaf-session.service"];
+        serviceConfig = {
+          # Previously there was "notify" type, but for some reason
+          # systemd kills weston.service because of timeout (even if it is disabled).
+          # "simple" works pretty well, so let's leave it.
+          Type = "simple";
+          #TimeoutStartSec = "60";
+          #WatchdogSec = "20";
+          # Defaults to journal
+          StandardOutput = "journal";
+          StandardError = "journal";
+          ExecStart = "${pkgs.weston}/bin/weston";
+          #GPU pt needs some time to start - weston fails to restart 3 times in avg.
+          ExecStartPre = "${pkgs.coreutils}/bin/sleep 3";
+          Restart = "on-failure";
+          RestartSec = "1";
+          # Ivan N: I do not know if this is bug or feature of NixOS, but
+          # when I add weston.ini file to environment.etc, the file ends up in
+          # /etc/xdg directory on the filesystem, while NixOS uses
+          # /run/current-system/sw/etc/xdg directory and goes into same directory
+          # searching for weston.ini even if /etc/xdg is already in XDG_CONFIG_DIRS
+          # The solution is to add /etc/xdg one more time for weston service.
+          # It does not affect on system-wide XDG_CONFIG_DIRS variable.
+          #
+          # Ivan N: adding openssh into the PATH since it is needed for waypipe to work
+          Environment = "XDG_CONFIG_DIRS=$XDG_CONFIG_DIRS:/etc/xdg PATH=${pkgs.openssh}/bin:$PATH";
+        };
+        wantedBy = ["default.target"];
       };
-      wantedBy = ["multi-user.target"];
-    };
-  };
+    })
+    (lib.mkIf cfg.enable (import ./window-manager.nix {inherit lib config pkgs;}))
```

Comment:
I think if you put all the stuff under same mkIf, you might not need the mkMerge above.

To Line 71, between } and ), you could put `// (import ./window-manager.nix {inherit lib config pkgs;})`

Actually at this point I realize, couldn't the window-manager.nix just have its own mkIf?

Reply:
Made it a separate module